tests/streams/test_memory.py

```diff
@@ -56,6 +56,17 @@ async def test_send_then_receive_nowait():
         assert await receive.receive_nowait() == 'hello'
 
 
+async def test_send_is_unblocked_after_receive_nowait():
+    send, receive = create_memory_object_stream(1)
+    await send.send_nowait('hello')
+    async with create_task_group() as tg:
+        await tg.spawn(send.send, 'anyio')
+        await wait_all_tasks_blocked()
+        assert await receive.receive_nowait() == 'hello'
+        await wait_all_tasks_blocked()
+        assert len(receive._state.buffer) == 1
```

Comment:
Rather than checking for the internal state of the buffer, I would prefer to have the task group surrounded with a `fail_after(1)` and then `assert receive.receive_nowait() == 'anyio'` outside of it.

Reply:
I believe that would pass in the current (broken) master, since the second `receive_nowait` works just fine. I think we need to test that the `send` is unblocked after the first `receive_nowait`.

I understand if you don't want to use internal state. Any other ideas?

Reply:
Oh, I think I was wrong, `fail_after` would will raise in master. I'll implement it.